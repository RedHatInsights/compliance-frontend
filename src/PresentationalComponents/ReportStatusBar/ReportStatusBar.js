import React from 'react';
import propTypes from 'prop-types';
import { Grid, GridItem } from '@patternfly/react-core';
import {
  global_BackgroundColor_light_300,
  global_palette_blue_200,
  global_palette_blue_400,
  global_palette_gold_300,
} from '@patternfly/react-tokens';
import MultiSegmentBar from '../MultiSegmentBar/MultiSegmentBar';

const ReportStatusBar = ({
  hostCounts: { compliant, totalResults, unsupported, total },
}) => {
  const counts = {
    compliant: compliant,
    nonCompliant: totalResults - compliant,
    unsupported: unsupported,
    neverReported: total - totalResults - unsupported,
  };

  const data = [
    {
      name: 'Compliant',
      color: global_palette_blue_400.var,
      value: compliant,
      label: `Compliant: ${counts.compliant}`,
    },
    {
      name: 'Non-Compliant',
      color: global_palette_blue_200.var,
      value: counts.nonCompliant,
      label: `Non-Compliant: ${counts.nonCompliant}`,
    },
    {
      name: 'Unsupported',
      color: global_palette_gold_300.var,
      value: counts.unsupported,
      label: `Unsupported: ${counts.unsupported}`,
    },
    {
      name: 'Never reported',
      color: global_BackgroundColor_light_300.var,
      value: counts.neverReported,
      label: `Never reported: ${counts.neverReported}`,
    },
  ];

  const compliancePercentage = totalResults
    ? (counts.compliant * 100) / totalResults
    : 0;

  return (
    <Grid hasGutter>
      <GridItem span={9}>
        <MultiSegmentBar data={data} />
      </GridItem>
      <GridItem span={3}>
        <span>{`${Math.round(compliancePercentage)}%`}</span>
      </GridItem>
    </Grid>
  );
};

ReportStatusBar.propTypes = {
  hostCounts: propTypes.exact({
    totalResults: propTypes.number,
    compliant: propTypes.number,
    unsupported: propTypes.number,
    total: propTypes.number,
  }),
};

export default ReportStatusBar;
