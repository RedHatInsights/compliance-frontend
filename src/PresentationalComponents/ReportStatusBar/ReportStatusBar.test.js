import { shallow } from 'enzyme';
import ReportStatusBar from './ReportStatusBar';
window.ResizeObserver = class {
  constructor() {}

  observe() {}
};

const defaultHostCounts = {
  compliant: 5,
  totalResults: 10,
  unsupported: 2,
  total: 15,
};

describe('ReportStatusBar', () => {
  it('expect to render without error', () => {
    const wrapper = shallow(<ReportStatusBar hostCounts={defaultHostCounts} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('expect to render without error with no reporting systems', () => {
    const hostCounts = {
      ...defaultHostCounts,
      compliant: 0,
      totalResults: 0,
    };

    const wrapper = shallow(<ReportStatusBar hostCounts={hostCounts} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
