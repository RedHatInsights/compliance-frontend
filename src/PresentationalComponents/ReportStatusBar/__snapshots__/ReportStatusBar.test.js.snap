// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportStatusBar expect to render without error 1`] = `
<Grid
  hasGutter={true}
>
  <GridItem
    span={9}
  >
    <MultiSegmentBar
      data={
        Array [
          Object {
            "color": "var(--pf-global--palette--blue-400)",
            "label": "Compliant: 5",
            "name": "Compliant",
            "value": 5,
          },
          Object {
            "color": "var(--pf-global--palette--blue-200)",
            "label": "Non-Compliant: 5",
            "name": "Non-Compliant",
            "value": 5,
          },
          Object {
            "color": "var(--pf-global--palette--gold-300)",
            "label": "Unsupported: 2",
            "name": "Unsupported",
            "value": 2,
          },
          Object {
            "color": "var(--pf-global--BackgroundColor--light-300)",
            "label": "Never reported: 3",
            "name": "Never reported",
            "value": 3,
          },
        ]
      }
    />
  </GridItem>
  <GridItem
    span={3}
  >
    <span>
      50%
    </span>
  </GridItem>
</Grid>
`;

exports[`ReportStatusBar expect to render without error with no reporting systems 1`] = `
<Grid
  hasGutter={true}
>
  <GridItem
    span={9}
  >
    <MultiSegmentBar
      data={
        Array [
          Object {
            "color": "var(--pf-global--palette--blue-400)",
            "label": "Compliant: 0",
            "name": "Compliant",
            "value": 0,
          },
          Object {
            "color": "var(--pf-global--palette--blue-200)",
            "label": "Non-Compliant: 0",
            "name": "Non-Compliant",
            "value": 0,
          },
          Object {
            "color": "var(--pf-global--palette--gold-300)",
            "label": "Unsupported: 2",
            "name": "Unsupported",
            "value": 2,
          },
          Object {
            "color": "var(--pf-global--BackgroundColor--light-300)",
            "label": "Never reported: 13",
            "name": "Never reported",
            "value": 13,
          },
        ]
      }
    />
  </GridItem>
  <GridItem
    span={3}
  >
    <span>
      0%
    </span>
  </GridItem>
</Grid>
`;
